backend/src/middleware/handleNotifications/handleNotificationsMiddleware.js

```diff
@@ -0,0 +1,156 @@
+import { UserInputError } from 'apollo-server'
+import extractMentionedUsers from './notifications/extractMentionedUsers'
+import extractHashtags from './hashtags/extractHashtags'
+
+const notifyUsers = async (label, id, idsOfUsers, reason, context) => {
+  if (!idsOfUsers.length) return
+
+  // Done here, because Neode validation is not working.
+  const reasonsAllowed = ['mentioned_in_post', 'mentioned_in_comment', 'comment_on_post']
+  if (!reasonsAllowed.includes(reason)) {
+    throw new UserInputError('Notification reason is not allowed!')
+  }
+  if (
+    (label === 'Post' && reason !== 'mentioned_in_post') ||
+    (label === 'Comment' && !['mentioned_in_comment', 'comment_on_post'].includes(reason))
+  ) {
+    throw new UserInputError('Notification fits not to reason!')
```

Comment:
this would be better `Notification does not fit the reason` and this would not be a `UserInputError`, right?
for this to happen, it would not be the user doing something incorrectly, or is this for someone hitting the backend directly and trying to create a notification without the correct reasons?

Reply:
Changed the translation.
Thanks for the a better one! 😍 @mattwr18 

Yap, you are absolutely right. It is just an `Error`. 👍🏼